File: IT/Programmazione funzionale/Esercizi/Java/Esercizio12.java
/* file Esercizio12.java */

import java.util.List;
import java.util.stream.IntStream;
import java.util.stream.Collectors;

public class Esercizio12 {
	public static void main(String [] args) {
		// trova tutti i numeri dispari che sono anche multipli di 3

		List<Integer> m = IntStream.rangeClosed(1, 100).filter(n -> n % 3 == 0).boxed().filter(n -> n % 2 == 1).collect(Collectors.toList());
		System.out.println(m);
	}
}

File: IT/2/Programmazione funzionale/Esercizi/Python/esercizio12.py
# file esercizio12.py

from __future__ import print_function

li = [n for n in range(1, 100+1)]

# trova tutti i numeri dispari che sono anche multipli di 3

m = list(filter(lambda n : n % 3 == 0, filter(lambda n : n % 2 == 1, li)))

print(m)

